controllers/gatekeeper_controller.go

```diff
@@ -962,14 +945,68 @@ func containerOverrides(obj *unstructured.Unstructured, spec operatorv1alpha1.Ga
 	return nil
 }
 
-// setReplicas
-func setReplicas(obj *unstructured.Unstructured, replicas *int32) error {
-	if replicas != nil {
-		if err := unstructured.SetNestedField(obj.Object, int64(*replicas), "spec", "replicas"); err != nil {
+// setCommonConfig takes a deployment and CommonConfig struct and applies the configuration from
+// CommonConfig to the deployment, returning any errors.
+func setCommonConfig(log logr.Logger, obj *unstructured.Unstructured, config operatorv1alpha1.CommonConfig) error {
+	// Set pod replicas
+	if config.Replicas != nil {
+		if err := unstructured.SetNestedField(obj.Object, int64(*config.Replicas), "spec", "replicas"); err != nil {
 			return errors.Wrapf(err, "Failed to set replica value")
 		}
 	}
 
+	// Set container resources
+	if config.Resources != nil {
+		err := setContainerAttrWithFn(obj, func(container map[string]interface{}) error {
+			if err := unstructured.SetNestedField(container, util.ToMap(config.Resources), "resources"); err != nil {
+				return errors.Wrapf(err, "Failed to set container resources")
+			}
+
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Set --log-level flag
+	if config.LogLevel != nil {
+		if err := setContainerArg(obj, LogLevelArg, string(*config.LogLevel)); err != nil {
+			return err
+		}
+	}
+
+	// Set --metrics-backend flag to Prometheus (this flag can be set more than
+	// once for additional metrics providers)
+	if err := setContainerArg(obj, "--metrics-backend", "prometheus"); err != nil {
```

Comment:
Is this a new flag for Gatekeeper 3.17?

Reply:
...but that was for the ability to specify multiple metrics backends. The `metrics-backend` flag has existed since 3.1:
- https://github.com/open-policy-agent/gatekeeper/commit/6d1c641d6a291ce0e39425905b2fd635cda5c34d